src/please-wait.coffee

```diff
@@ -245,6 +247,7 @@
         if animationSupport then @_loadingElem.removeEventListener(animationEvent, listener)
         # Reset the loading screen element since it's no longer attached to the DOM
         @_loadingElem = null
+        @_onFinishedCallback.apply(this)
```

Comment:
Yes, I ended up creating a separate repository instead, and I added a null check there. Thanks for the helpful response, I have a feeling I’m not quite following the correct protocol here.

The reason behind why I added a second callback was that I’m developing an SPA that uses google maps extensively. Unfortunately, the google maps canvas won’t size properly unless it is displayed prior to loading. So I needed a callback that I knew would only be fired once my content was displayed and pg-loaded had been set on the body.

Thanks again for your patience and for the excellent job you guys have done with please-wait.

-Jonathan

Från: Neville Samuell [mailto:notifications@github.com]
Skickat: den 11 mars 2016 14:34
Till: Pathgather/please-wait <please-wait@noreply.github.com>
Kopia: Jonathan Quinth <jonathan@iosoft.se>
Ämne: Re: [please-wait] Added callback when loading screen has finished fading out (#21)


In src/please-wait.coffee<https://github.com/Pathgather/please-wait/pull/21#discussion_r55828853>:

> @@ -245,6 +247,7 @@

>          if animationSupport then @_loadingElem.removeEventListener(animationEvent, listener)

>          # Reset the loading screen element since it's no longer attached to the DOM

>          @_loadingElem = null

> +        @_onFinishedCallback.apply(this)

What about if the user doesn't provide the callback option?

—
Reply to this email directly or view it on GitHub<https://github.com/Pathgather/please-wait/pull/21/files#r55828853>.